Automattic/jetpack
# Protect: poll status while "scheduled"

This PR adds a status polling component to the Protect admin page. When the status is "scheduled", the page will re-fetch and re-check the status every 10 seconds until the status is no longer "scheduled".

The purpose of this is to automatically update the page with the latest Protect results when viewing the page after a fresh installation of the plugin. 

#### Changes proposed in this Pull Request:

* Adds and implements a `useStatusPolling()` hook to `<Admin />`. When the `status` is "scheduled", a timeout is run to re-check the status every 10 seconds until the value changes.

#### Jetpack product discussion



1201069996155224-as-1202285393876621/f

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Activate Protect on a fresh JN site running this branch.
* Open the Admin page and validate that the "Your results will be ready soon..." message is shown.
* Validate that the status request is firing every 10 seconds in the Network tab.
* Wait!
* Validate that the scan results are displayed when they become available when the status response is no longer "scheduled".
